pyOCD/board/mbed_board.py

```diff
@@ -150,7 +150,8 @@ def listConnectedBoards(dap_class=DAPAccess):
 
     @staticmethod
     def getAllConnectedBoards(dap_class=DAPAccess, close=False, blocking=True,
-                              target_override=None, frequency=1000000):
+                              target_override=None, frequency=1000000,
+                              board_id = None):
```

Comment:
Yeah, that seems like a reasonable limitation.  At least for DAPLink each board is guaranteed to have a unique serial number.